### DISCUSSION TITLE
Notifications/message don't display if status bar is turned off

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#1891] [Q&A] [answered]
I am using the following call to output messages (taken from frontend-samples-showcase)

IModelApp.notifications.outputMessage(new NotifyMessageDetails(OutputMessagePriority.Info, "Hello"));

We are not using status bar in our app so it is turned off using defaultUiConfig parameter of <Viewer> class

...
  hideDefaultStatusBar: true
...

It seems that if status bar is turned off, messages don't show up.  If I set the above to hideDefaultStatusBar to 'false', messages start appearing as expected (status bar is also shown which is what we don't want).  

Can you please guide how we can show our notifications without having to turn on the status bar.  Perhaps I can set these listeners myself (pls see below...)

In trying to trace this issue, it seems certain listeners are not set if status bar is off.  I have tried to capture the call stack below.

@bentley/bentleyjs-core/src/BeEvents
public raiseEvent(...args: Parameters<T>) 
...

If status bar is off
[[FunctionLocation]]: => blank/undefined
listeners.length => 0 

If status bar is on
[[FunctionLocation]]: => set to @bentley\ui-framework\statusbar\StatusBar.tsx:137
listeners.length => 1 

messages keep stacking up in queue and not displayed

Called from:

@bentley/bentleyjs-core/src/BeEvents
export class BeUiEvent<TEventArgs> extends BeEvent<(args: TEventArgs) => void> {
  /** Raises event with single strongly typed argument. */
  public emit(args: TEventArgs): void { this.raiseEvent(args); }
}

Called from @bentley/ui-framework/messages/MessageManager.ts

public static addMessage(message: NotifyMessageDetailsType): void {
...
   this.onMessageAddedEvent.emit({ message });
...
}

Called from @bentley/ui-framework/messages/AppNotificationManager.ts

  public outputMessage(message: NotifyMessageDetails): void {
    MessageManager.outputMessage(message);
  }

**Comment by DanEastBentley** [comment, marked as answer]
If you add &lt;MessageRenderer /&gt; from "@bentley/ui-framework" to your React component hierarchy, you will see the messages without using the StatusBar.

**Reply by ovaisa** [reply]
Thank you Dan, I can now see notifications as expected.
